validateUserInput.py

```
@@ -0,0 +1,142 @@
+		# pattern object for invalidChars
+		nameSecondParent_noSpace = re.sub(r'\s+', '', nameSecondParent)
+		m_nameSecondParent = p_invalidChars.search(nameSecondParent_noSpace)
+		if m_nameSecondParent:
+			raise Exception('there are either non-alphanumeric characters, underscore, or digits in nameSecondParent')
+
+	return namePerson, birthDate, nameFirstParent, nameSecondParent
+
+def deathNoticeValidate(userInputList):
+	# deathNotice method: <name-of-deceased>, [<optional-date-of-death>]
+	# Ex: deathNotice, David Bowie, 01/10/2016
+	# Note about deathNotice: must be called with name of existing Person object
+	# Note about deathNotice: date-of-death is optional because might be unknown
+	if len(userInputList) == 2:
+		_, nameDeceased = userInputList
+		# default of deathDate is None if not omitted
+		deathDate = None
+	elif len(userInputList) == 3:
+		_, nameDeceased, deathDate = userInputList
+	
+	# remove leading and trailing whitespace
+	nameDeceased.strip()
+	try:
+		deathDate.strip()
+	except AttributeError:
+		pass
+
+	# pattern object for invalidChars
+	nameDeceased_noSpace = re.sub(r'\s+', '', nameDeceased)
+	p_invalidChars = re.compile(r'[\W_\d]', re.I | re.X)
+	m_nameDeceased = p_invalidChars.search(nameDeceased_noSpace)
+	if m_nameDeceased:
+		raise Exception('there are either non-alphanumeric characters, underscore, or digits in nameDeceased')
+
+	if deathDate is not None:
+		# pattern object for deathDate_format
+		deathDate_noSpace = re.sub(r'\s+', '', deathDate)
+		p_deathDate_format = re.compile(r'\d\d/\d\d/\d\d\d\d')
+		m_deathDate_format = p_deathDate_format.match(deathDate_noSpace)
+		if not m_deathDate_format:
+			raise Exception('deathDate is not in the proper format: mm/dd/yyyy')
+
+	return nameDeceased, deathDate
+
+def displayTreeValidate(userInputList):
+	# displayTree method takes no arguments')
+	# Ex: displayTree
+	if len(userInputList) > 1:
+		raise Exception('displayTree method takes no arguments')
+#
```

Better to fail explicitly than fail silently.